js/nhsuk.js

```diff
@@ -1311,3 +1311,45 @@ document.querySelectorAll(".nhsuk-card--clickable").forEach((panel) => {
 		});
 	}
 });
+
+const toggleButton = document.querySelector('#toggle-menu');
+const closeButton = document.querySelector('#close-menu');
+const nav = document.querySelector('#header-navigation');
+toggleButton.addEventListener("click", () => {
+	//nav.addEventListener('transitionend', (e) => {
+	if (toggleButton.classList.contains("is-active") === false) { // weird thing where we have to negatively search for
+		// the active indicator, as at the point of searching the flag is not set. So it only exists when we are
+		// actually hiding the nav
+		setTimeout(function(){
+			closeButton.focus();
+			trapFocus(nav); // put the keyboard nav in a loop inside the menu element.
+		},100); // add in a slight pause so the nav actually displays and becomes focussable
+	}
+	// });
+});
+function trapFocus(element) {
```

Comment:
Considering that code is not ours ...see (www.taraprasad.com) and heavy I'd suggest using https://css-tricks.com/a-css-approach-to-trap-focus-inside-of-an-element/ which is way cleaner, safer and does not have a heavy code smell behind it